src/main/java/com/amplitude/api/AmplitudeClient.java

```diff
@@ -2047,36 +2018,57 @@ protected void makeEventUploadPostRequest(Call.Factory client, String events, fi
             logger.e(TAG, e.toString());
         }
 
-        FormBody body = new FormBody.Builder()
-            .add("v", apiVersionString)
-            .add("client", apiKey)
-            .add("e", events)
-            .add("upload_time", timestampString)
-            .add("checksum", checksumString)
-            .build();
+        HttpsURLConnection connection = null;
 
-        Request request;
         try {
```

Comment:
Better to extract the logic into a separate class.
https://github.com/mixpanel/mixpanel-android/blob/2766eab480eef892a044f7fb2673453919661c11/src/main/java/com/mixpanel/android/util/HttpService.java